core/src/main/java/org/elasticsearch/ingest/core/Processor.java

```diff
@@ -49,7 +54,6 @@
          * Implementations are responsible for removing the used keys, so that after creating a pipeline ingest can
          * verify if all configurations settings have been used.
          */
-        P create(Map<String, Object> config) throws Exception;
-
+        P create(String tag, Map<String, Object> config) throws Exception;
```

Comment:
Sorry I meant `AbstractProcessorFactory`. Something like this:

```java
public abstract class AbstractProcessorFactory implements Processor.Factory {

    @Override
    public Processor create(Map config) throws Exception {
        String tag = ConfigurationUtils.readOptionalStringProperty(config, "tag");
        return doCreate(tag, config);
    }
    
    protected abstract Processor doCreate(String tag, Map config) throws Exception;
    
}
```

All concrete processor would extend from that instead of `Processor.Factory`.  Since the tag is an optional implementation detail, it feels nicer if that isn't visible in the interface, but is moved to the concrete processor implementations. You think so too?